mcmc_utils.py

```diff
@@ -280,9 +286,49 @@ def flatchain(chain, npars, nskip=0, thin=1):
     the steps of the chain are highly correlated'''
     return chain[:, nskip::thin, :].reshape((-1, npars))
 
-
-def readchain(file, nskip=0, thin=1):
-    data = pd.read_csv(file, header=None, compression=None, delim_whitespace=True)
+def reverse_readline(filename, buf_size=8192):
```

Comment:
Interesting idea this, but I'm not sure of its utility. You *would* normally want to read the whole chain, and sample from it if it's too big. The end is not more important than the beginning, unless you are not burnt-in sufficiently...

Reply:
In that case, I'll discard this change.